### ISSUE TITLE
Various GraphQL Transform issues / inconsistencies

### ISSUE THREAD

**Issue by dabit3** [aws-amplify/amplify-cli#1968]
## Issue 1:

**Is your feature request related to a problem? Please describe.**
Yes, when creating an `@model` and specifying a sort key, the `id` and `createdAt` values are no longer auto-generated by the resolver.

**Describe the solution you'd like**
Right now, here is the request mapping template created based on the following GraphQL schema for a mutation:

Schema:
```graphql
type Talk @model @key(fields: ["id", "createdAt"]) {
  id: ID!
  name: String!
  description: String
  createdAt: String!
}
```

Resolver (`createTalk`):
```vtl
## [Start] Set the primary @key. **
#set( $modelObjectKey = {
  "id": $util.dynamodb.toDynamoDB($ctx.args.input.id),
  "createdAt": $util.dynamodb.toDynamoDB($ctx.args.input.createdAt)
} )
## [End] Set the primary @key. **

## [Start] Prepare DynamoDB PutItem Request. **
$util.qr($context.args.input.put("createdAt", $util.defaultIfNull($ctx.args.input.createdAt, $util.time.nowISO8601())))
$util.qr($context.args.input.put("updatedAt", $util.defaultIfNull($ctx.args.input.updatedAt, $util.time.nowISO8601())))
$util.qr($context.args.input.put("__typename", "Talk"))
```

Instead of expecting and `id` and `createdAt` value, we should also assume that they may want these values auto-generated by the resolver:

```vtl
## [Start] Set the primary @key. **
#set( $modelObjectKey = {
  "id": $util.dynamodb.toDynamoDB($util.defaultIfNullOrBlank($ctx.args.input.id, $util.autoId())),
  "createdAt": $util.dynamodb.toDynamoDB($util.defaultIfNull($ctx.args.input.createdAt, $util.time.nowISO8601()))
} )
## [End] Set the primary @key. **

## [Start] Prepare DynamoDB PutItem Request. **
$util.qr($context.args.input.put("createdAt", $util.defaultIfNull($ctx.args.input.createdAt, $util.time.nowISO8601())))
$util.qr($context.args.input.put("updatedAt", $util.defaultIfNull($ctx.args.input.updatedAt, $util.time.nowISO8601())))
$util.qr($context.args.input.put("id", $util.defaultIfNull($ctx.args.input.id, $util.autoId())))
$util.qr($context.args.input.put("__typename", "Talk"))
```

## Issue 2:
We should also assume that the `id` and `createdAt` values will be an optional / nullable field in the input type for a mutation:

I.e. when I create this Type using the GraphQL transform *without* the @key argument, the input type has these fields marked as optional because the resolver will do a null check and then create them if they are not included in the arguments or input object.

```graphql
input CreateTalkInput {
  id: ID! $ <- this should be nullable
  name: String!
  description: String
  createdAt: String! <- this should be nullable
}
```

cc @kaustavghosh06 @undefobj

**Comment by attilah** [comment]
@dabit3 Let me reopen it to make sure we will have this covered later, just un-assigning myself as it is an enhancement.

**Comment by attilah** [comment]
@dabit3 Issue 1 is fixed by @yuth in #4149

**Comment by undefobj** [comment]
These should be resolved now.
